cpp/src/mlt/geometry_vector.cpp

```diff
@@ -0,0 +1,545 @@
+#include <mlt/geometry_vector.hpp>
+
+#include <mlt/coordinate.hpp>
+#include <mlt/polyfill.hpp>
+#include <mlt/util/morton_curve.hpp>
+#include <mlt/util/stl.hpp>
+
+namespace mlt::geometry {
+
+#if !defined(MLT_GEOMETRY_CHECKS)
+#define MLT_GEOMETRY_CHECKS 1
+#endif
+namespace {
+#if MLT_GEOMETRY_CHECKS
+void checkBuffer(std::size_t index, std::size_t size, std::string_view name) {
```

Comment:
Suggestion: make this an inline function with if constexpr, then the macro is not needed.